itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosExternalTables.java

```diff
@@ -750,6 +766,38 @@ public Table apply(@Nullable Table table) {
             .verifyResults(Arrays.asList("1", "2"));
   }
 
+  @Test
+  public void testIncrementalDumpEmptyDumpDirectory() throws Throwable {
```

Comment:
Add another test case where we dynamically bootstrap a table (table level replication) with incremental dump but no events are dumped. It takes a special route in executeIncrementalLoad() method Line: 503 and so I guess, as per current change, it won't update the database last repl ID.

Reply:
I have added a testcase in the patch. It's passing for me. Can you please check the same?

Reply:
We cannot reproduce this scenario with acid enabled. If ACID enabled, we will have at least open/commit txn event part of each incremental dump. So, impossible to get empty incremental dump. Please try this scenario with ACID disabled.

Reply:
Added a testcase with external table bootstrap. There I could reproduce the problem you mentioned and also fixed it. During an incremental the last repl id is updated for the database after applying all the events but before bootstrapping any tables.